lib/shared/translate.js

```diff
@@ -0,0 +1,267 @@
+    }
+    case messages.FLAG_TASKS_JSON: {
+      return chalk.gray(
+        'Print the task dependency tree, ' +
+        'in JSON format, for the loaded gulpfile.'
+      );
+    }
+    case messages.FLAG_TASKS_DEPTH: {
+      return chalk.gray('Specify the depth of the task dependency tree.');
+    }
+    case messages.FLAG_COMPACT_TASKS: {
+      return chalk.gray(
+        'Reduce the output of task dependency tree by printing ' +
+        'only top tasks and their child tasks.'
+      );
+    }
+    case messages.FLAG_SORT_TASKS: {
+      return chalk.gray('Will sort top tasks of task dependency tree.');
+    }
+    case messages.FLAG_COLOR: {
+      return chalk.gray(
+        'Will force gulp and gulp plugins to display colors, ' +
+        'even when no color support is detected.'
+      );
+    }
+    case messages.FLAG_NO_COLOR: {
+      return chalk.gray(
+        'Will force gulp and gulp plugins to not display colors, ' +
+        'even when color support is detected.'
+      );
+    }
+    case messages.FLAG_SILENT: {
+      return chalk.gray('Suppress all gulp logging.');
+    }
+    case messages.FLAG_CONTINUE: {
+      return chalk.gray('Continue execution of tasks upon failure.');
+    }
+    case messages.FLAG_SERIES: {
+      return chalk.gray('Run tasks given on the CLI in series (the default is parallel).');
+    }
+    case messages.FLAG_LOG_LEVEL: {
+      return chalk.gray(
+        'Set the loglevel. -L for least verbose and -LLLL for most verbose. ' +
+        '-LLL is default.'
+      );
+    }
+    case messages.BOX_DRAWINGS_LIGHT_UP_AND_RIGHT: {
+      return chalk.white('└');
+    }
+    case messages.BOX_DRAWINGS_LIGHT_VERTICAL_AND_RIGHT: {
+      return chalk.white('├');
+    }
+    case messages.BOX_DRAWINGS_LIGHT_HORIZONTAL: {
+      return chalk.white('─');
+    }
+    case messages.BOX_DRAWINGS_LIGHT_DOWN_AND_HORIZONTAL: {
+      return chalk.white('┬');
+    }
+    case messages.BOX_DRAWINGS_LIGHT_VERTICAL: {
+      return chalk.white('│');
+    }
+    default: {
+      return msg;
+    }
+  }
+}
+
+function getDefaultTimestamp() {
+  return util.inspect(new Timestamp(), { colors: !!chalk.supportsColor });
+}
+
+function buildTranslations(cfg) {
+  cfg = cfg || {};
+
+  return {
+    message: function (msg, data) {
+      var message;
+      if (typeof cfg.message === 'function') {
+        message = cfg.message(msg, data);
+      }
+
+      // If the user has provided a message, return it
+      if (message) {
+        return message;
+      }
+
+      // Users can filter messages by explicitly returning `false`
+      if (message === false) {
+        return '';
+      }
+
+      // If the user hasn't returned a message or silenced it with `false`
+      // get the default message. Will return the first argument if the message
+      // is not defined in the `@gulpjs/messages` package
+      return getDefaultMessage(msg, data);
+    },
+    timestamp: function () {
```

Comment:
I considered passing the message symbol here, but I think that we just need to separate timestamp vs non-timestamp output by the usage of `log` vs `console.log`

Reply:
I think it's better that `message` function always returns a line and receives message parts via `data` parameter, and timestamp is passed by including `data` parameter, too.
By doing so, we and users can choose whether attaching timestamp before a message line in the`message` function.

Reply:
I don't want to pass message parts but I'm considering attaching the timestamp in the `message` function. Please let me consider which is preferable.

Reply:
I've been considering your proposal, but I was struggling with some ideas.

I want users to be able to easily disable timestamps if they don't like them. If we attach timestamps in the `message` function, users would have to override all messages to disable timestamps.

Instead, I've decided to pass the message into the `timestamp` function so users can vary the timestamp based on the message. However, people are able to ignore the parameters and return a single timestamp.

Reply:
There is no way for the user to access getDefaulMessage. If they return nothing, the default message is used. If they return `false`, the message is silenced.

Reply:
Sorry. What I wanted to do in the second example above can be written as follows.
```
    getMessage = function(data) {
        switch (data.tag) {
        case messages.PRELOAD_BEFORE:
            return false;   // message isn't displayed.
        case messages.MISSING_GULPFILE:
            return timestamp() + chalk.yellow('Gulpfile was not found!')  + ' 😱😱😱' ;
        default:
            return ''; // the default message is displayed.
        }
    }
```

Reply:
What I really wanted to suggest is passing the `timestamp` function to the `data`, and then in the message function, appending it (or removing it if the user doesn't want it).

Reply:
I don't want to attach timestamp to the data object. I don't want to require users to implement the `message()` function if they only want to remove the timestamp.

Reply:
To be clear, I want it to be 1-3 lines of JS to disable timestamps. With this implementation that is possible.

Reply:
Understood, I think that's fine.